### PR TITLE
Fix 947

### PR DESCRIPTION
**Pull request by antgonza**
This adds a nicer visualization of dropped samples in an analysis. This is the base to add https://github.com/biocore/qiita/issues/1091, which will be in another PR.

### CODE REVIEW

File: qiita_pet/handlers/analysis_handlers.py [hunk 1]
```diff
@@ -141,14 +142,19 @@ def get(self, analysis_id):
             jobres[jobject.datatype].append((jobject.command[0],
                                              jobject.results))
 
-        dropped = {}
         dropped_samples = analysis.dropped_samples
+        dropped = {}
```
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers.py]: You can use `defaultdict(list)` and remove the `if` statement in lines 154-155

File: qiita_pet/templates/analysis_results.html [hunk 2]
```diff
@@ -11,44 +11,60 @@
...
+          </div>
+          <div id="dropped-samples" class="panel-collapse collapse">
+              <div class="panel-body">
+                  {% for data_type, studies in viewitems(dropped) %}
+                    <b><u>{{data_type}}</u></b>
+                    <br/><br/>
+                    {% for s in studies %}
+                      <b>{{s['study']}}:</b>
+                      <br/>
+                      Total dropped: {{len(s['samples'])}}
```
Comment (josenavas) [inline qiita_pet/templates/analysis_results.html]: Can you compute the length of the list and do the `', '.join` in the handler? In previous iterations of the code we avoid to add as much python code in templates as possible.

File: qiita_pet/templates/analysis_results.html [hunk 3]
```diff
@@ -11,44 +11,60 @@
...
+        <div class="panel panel-default">
+          <div class="panel-heading">
+            <h3 class="panel-title"><a data-toggle="collapse" data-parent="#accordion" href="#dropped-samples">Dropped Samples</a></h3>
+          </div>
+          <div id="dropped-samples" class="panel-collapse collapse">
+              <div class="panel-body">
+                  {% for data_type, studies in viewitems(dropped) %}
+                    <b><u>{{data_type}}</u></b>
+                    <br/><br/>
+                    {% for s in studies %}
```
Comment (josenavas) [inline qiita_pet/templates/analysis_results.html]: I think it is better to use tuples rather than dictionaries in here:
```
{% for title, num_samples, samples in studies %}
```
this helps to reduce also the amount of python code.

**Comment by josenavas** [comment]
Thanks @antgonza this looks way better! Few comments though...

**Comment by josenavas** [comment]
:+1:

File: qiita_pet/handlers/analysis_handlers.py [hunk 4]
```diff
@@ -141,14 +142,17 @@ def get(self, analysis_id):
             jobres[jobject.datatype].append((jobject.command[0],
                                              jobject.results))
 
-        dropped = {}
         dropped_samples = analysis.dropped_samples
+        dropped = defaultdict(list)
         if dropped_samples:
-            for proc_data_id, samples in viewitems(dropped_samples):
+            for proc_data_id, samples in viewitems(analysis.dropped_samples):
```
Comment (adamrp) [inline qiita_pet/handlers/analysis_handlers.py]: Why did you change this line? `dropped_samples` was already set to `analysis.dropped_samples` on line 145

Comment (adamrp) [inline qiita_pet/handlers/analysis_handlers.py]: In fact, the `if` statement on line 147 is not needed here, since this `for` loop will not run if `dropped_samples` is empty

**Comment by adamrp** [comment]
A couple small comments

File: qiita_pet/handlers/analysis_handlers.py (hunk 4 above)
Comment (antgonza) [inline qiita_pet/handlers/analysis_handlers.py]: 1. analysis.dropped_samples -> dropped_samples: will change, left behind by mistake
2. if statement is necessary as dropped_samples is either a dict or None.

Thanks!

File: qiita_pet/handlers/analysis_handlers.py [hunk 5]
```diff
@@ -141,14 +142,17 @@ def get(self, analysis_id):
             jobres[jobject.datatype].append((jobject.command[0],
                                              jobject.results))
 
-        dropped = {}
         dropped_samples = analysis.dropped_samples
+        dropped = defaultdict(list)
         if dropped_samples:
```
Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers.py]: @adamrp I think you suggested that this `if` statement can be removed. However, `analysis.dropped_samples` returns `None` if there are no dropped samples, instead of an empty list. This will make the call to `viewitems` to fail because of an AttributeError.

I think this should not block this PR and we should open a new issue for modifying `analysis.dropped_samples` so it returns consistent types (list). Do you agree?

Comment (adamrp) [inline qiita_pet/handlers/analysis_handlers.py]: Agreed -- thanks for pointing that out, I didn't look carefully at `analysis.dropped_samples` before commenting.

Comment (josenavas) [inline qiita_pet/handlers/analysis_handlers.py]: Cool! New issue created: https://github.com/biocore/qiita/issues/1152
